core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala

```diff
@@ -561,6 +565,31 @@ final class EventHubsConf private (private val connectionStr: String)
             s"${PartitionPreferredLocationStrategy.values.mkString(",")}"))
   }
 
+  /**
+   * Use AAD auth to connect eventhubs instead of connection string.
+   * More info about this: https://docs.microsoft.com/en-us/azure/event-hubs/authorize-access-azure-active-directory
+   * Default: [[false]]
+   * @return the updated [[EventHubsConf]] instance
+   */
+  private def setUseAadAuth(b: Boolean): EventHubsConf = {
+    set(UseAadAuthKey, b)
+  }
+
+  def useAadAuth: Boolean = {
+    self.get(UseAadAuthKey).getOrElse(DefaultUseAadAuth).toBoolean
+  }
+
+  def setAadAuthCallback(callback: AadAuthenticationCallback): EventHubsConf = {
```

Comment:
Can you please add API summary for new APIs?

Reply:
sure, done.